importer/Bosnia.py

```diff
@@ -0,0 +1,321 @@
+        """
+        self.wd_item = {}
+        self.wd_item["upload"] = True
+        self.wd_item["statements"] = {}
+        self.wd_item["labels"] = {}
+        self.wd_item["aliases"] = {}
+        self.wd_item["descriptions"] = {}
+        self.wd_item["disambiguators"] = {}
+        self.wd_item["wd-item"] = None
+        self.mapping = mapping
+        self.create_source()
+        self.set_country()
+        self.set_adm_location()
+        self.set_coords()
+        self.set_is()
+        self.set_heritage()
+        self.set_described()
+        self.set_street_address()
+        self.update_descriptions()
+        self.update_labels()
+
+
+def generate_filenames(tablename, timestamp):
+    """Construct the problem and preview filenames."""
+    filenames = {}
+    utils.create_dir(REPORTING_DIR)
+    filenames['reports'] = os.path.join(
+        REPORTING_DIR, "report_{0}_{1}.json".format(tablename, timestamp))
+
+    utils.create_dir(PREVIEW_DIR)
+    filenames['examples'] = os.path.join(
+        PREVIEW_DIR, "examples_{0}_{1}.wiki".format(tablename, timestamp))
+    filenames['matches'] = os.path.join(
+        PREVIEW_DIR, "matches_{0}_{1}.wiki".format(tablename, timestamp))
+    return filenames
+
+
+def load_source_data(directory):
+    """Load multiple source files from directory."""
+    source_data = []
+    for path, subdirs, files in os.walk(directory):
+        for name in files:
+            rows = utils.get_data_from_csv_file(os.path.join(path, name))
+            for row in rows:
+                source_data.append(row)
+    print("Loaded {} files, {} data rows.".format(len(files),
+                                                  len(source_data)))
+    return source_data
+
+
+def get_wd_items_using_prop(prop):
+    """
+    Get WD items that already have some value of a unique ID.
+
+    Even if there are none before we start working,
+    it's still useful to have in case an upload is interrupted
+    and has to be restarted, or if we later want to enhance
+    some items. When matching, these should take predecence
+    over any hardcoded matching files.
+
+    The output is a dictionary of ID's and items
+    that looks like this:
+    {'4420': 'Q28936211', '2041': 'Q28933898'}
+    """
+    items = {}
+    print("WILL NOW DOWNLOAD WD ITEMS THAT USE " + prop)
+    query = "SELECT DISTINCT ?item ?value  WHERE {?item p:" + \
+        prop + "?statement. OPTIONAL { ?item wdt:" + prop + " ?value. }}"
+    data = lookup.make_simple_wdqs_query(query, verbose=False)
+    for x in data:
+        key = lookup.sanitize_wdqs_result(x['item'])
+        value = x['value']
+        items[value] = key
+    print("FOUND {} WD ITEMS WITH PROP {}".format(len(items), prop))
+    return items
+
+
+def load_mapping_files():
+    """Load the files with mappings of various values."""
+    mapping_files = {}
+    mapping_files["_static"] = {}
+    files_to_get = ["ba_(bs).json", "bosnia_adm.json", "bosnia_types.json"]
```

Comment:
`ba_(bs).json` is missing from the patch

Reply:
:white_check_mark: